CMakeLists.txt

```
@@ -55,30 +67,93 @@ set(SRC_FILES
     src/egm_logger.cpp
     src/egm_udp_server.cpp
     src/egm_trajectory_interface.cpp
-    ${EgmProtoSources})
-
-include_directories(include
-                    ${CMAKE_CURRENT_BINARY_DIR} # Contains protobuf generated sources
-                    ${Boost_INCLUDE_DIRS}
-                    ${PROTOBUF_INCLUDE_DIRS}
-                    ${catkin_INCLUDE_DIRS})
+    ${EgmProtoSources}
+)
 
 add_library(${PROJECT_NAME} ${SRC_FILES})
-target_include_directories(${PROJECT_NAME} PUBLIC ${CMAKE_CURRENT_BINARY_DIR})
-target_link_libraries(${PROJECT_NAME} ${Boost_LIBRARIES} ${PROTOBUF_LIBRARIES} ${catkin_LIBRARIES})
+target_include_directories(${PROJECT_NAME} PUBLIC
+  "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include;${CMAKE_CURRENT_BINARY_DIR}>"
+  $<INSTALL_INTERFACE:include>
+  PUBLIC ${Boost_INCLUDE_DIRS}
+  PUBLIC ${PROTOBUF_INCLUDE_DIRS}
+)
+target_link_libraries(${PROJECT_NAME} PUBLIC
+  ${Boost_LIBRARIES}
+  ${PROTOBUF_LIBRARIES}
+)
+
+if(BUILD_SHARED_LIBS)
+  # Causes the visibility macros to use dllexport rather than dllimport,
+  # which is appropriate when building the dll but not consuming it.
+  target_compile_definitions(${PROJECT_NAME} PRIVATE "ABB_LIBEGM_BUILDING_SHARED_LIBRARY")
+else()
+  target_compile_definitions(${PROJECT_NAME} PRIVATE "ABB_LIBEGM_STATIC_LIBRARY")
```

```suggestion
  target_compile_definitions(${PROJECT_NAME} PUBLIC "ABB_LIBEGM_STATIC_LIBRARY")
```

`ABB_LIBEGM_STATIC_LIBRARY` needs to be defined also for the compilation unit that consume the library, otherwise static structure will be decorated with spurious `__declspec(dllimport)` .